atsign-foundation/at_server
# feat: at_root_server: rate limiting for invalid atSigns; misc logging and other cleanup

**- What I did**
- Resolve https://github.com/atsign-foundation/at_server/issues/1590
- Resolve https://github.com/atsign-foundation/at_server/issues/1615
- Partially resolve https://github.com/atsign-foundation/at_server/issues/1599
  - @cpswan This PR implements a variation on item 1 in #1599 - see "How to verify it" below
- Miscellaneous other logging cleanup
- update and pin dependencies
- bump version to 2.1.0 and update changelog

**- How I did it**
- See this PR's commits

**- How to verify it**
- https://github.com/atsign-foundation/at_server/issues/1615
  - Run local root server (see `at_redis` and `at_root` scripts in `tools/run_locally` directory in this repo)
    - Server should start up without delay if the certificates are present
- https://github.com/atsign-foundation/at_server/issues/1590
  - Open a plain socket connection to it then disconnect (e.g. by `curl http://vip.ve.atsign.zone:64`)
    - No log message should be generated
- https://github.com/atsign-foundation/at_server/issues/1599
  - Open a TLS connection (e.g. via `openssl s_client -connect vip.ve.atsign.zone:64`)
    - Look up a non-existent atSign: the client will receive a 'null' response
    - Do the lookup again: after a 3-second delay, the client will receive a 'null' response
    - Do the lookup again: after a 3-second delay, the client connection will be terminated
